### ISSUE TITLE
Running Orleans in SF on Azure

### ISSUE THREAD

**Issue by miker1423** [ReubenBond/OrleansFabricSilo#9]
I recently tried to host Orleans in Service Fabric in Azure, I just followed what is said in this repo and Im having trouble. The service which is hosting the silo sometimes shows as it is in a OK state but then, after a few seconds it shows ERROR state, and this is the error showing "Error event: SourceId='System.FM', Property='State'. Partition is below target replica or instance count.", that is all the information I have, can someone help me?

**Comment by ReubenBond** [comment]
I should update the readme. 
Do you have these endpoints in your `ServiceManifest.xml`?

```
      <Endpoint Name="OrleansSiloEndpoint"  Type="Internal" Protocol="tcp"/>
      <Endpoint Name="OrleansProxyEndpoint"  Type="Internal" Protocol="tcp"/>
```

**Comment by miker1423** [comment]
yes, I have both.

**Comment by ReubenBond** [comment]
Could you paste your logs?

**Comment by miker1423** [comment]
I dont really know were they are stored... First time on cloud.

**Comment by ReubenBond** [comment]
No problems. Does it work locally in the emulator?
EDIT: by the way, we might be able to resolve this faster in chat. Feel free to DM me on Gitter [here](https://gitter.im/ReubenBond) or via the Orleans room: https://gitter.im/dotnet/orleans

**Comment by miker1423** [comment]
Yes, with no problem at all.

**Comment by stupiduglyfool** [comment]
I've also been experimenting with this on service fabric on azure, everything works fine in the local emulator environment, but when trying to publish to azure it appears that despite setting a dataconnection string, it is still trying to use 127.0.0.1.

I modified my local environment to use the azure connection, and debugging confirms its trying to use 127.0.01.

I've just grabbed the latest OrleansFabricSilo code from github, and it now seems to work..

**Comment by ReubenBond** [comment]
@stupiduglyfool are you saying that the nuget package and the github code are behaving differently? That's odd, since the nuget package was built on the latest source

**Comment by stupiduglyfool** [comment]
@ReubenBond I was using nuget version 1.20, as this is the current version of orleans we are using. This version always seemed to want to use development storage, even though I've configured a different storage provider.. and this yields an error similar to what @miker1423 described when I tried to host on azure service fabric, or local service fabric without the storage emulator running.

I grabbed the latest code for OrleansFabricSilo from github instead of relying on the v1.20 nuget package, and it works perfectly.
